### PR TITLE
feat(cards): add the shared non-scalar card option controls

### PR DESCRIPTION
**Pull request by fx**
## Summary

The three shared non-scalar config controls — the final task of `docs/changes/0014-universal-card-options.md`, which this PR marks **complete**.

The change document is emphatic that it must not be marked complete without them, and the reason is concrete: changes 0016, 0021, 0024 and 0026 all assume these controls exist rather than inventing one apiece. Nothing consumes them yet by design; those changes bind them to their keys.

| Control | Type | First consumer |
|---|---|---|
| Entity picker | `entity` | `motionEntity` / `doorEntity` / `batteryEntity` (0021, 0024, 0026) |
| Number array | `number-array` | `brightnessPresets` (0016) |
| Ordered multi-select | `ordered-multi-select` | `armModes` (0024) |

Strict schemas live in `src/store/configControls.ts` for the import gate, deliberately split from the readers used on the render path.

## Forward compatibility, verified rather than asserted

`docs/specs/dashboard-config/index.md` requires that a configuration this build cannot fully interpret survives a round-trip unchanged — resolve for display, never rewrite or truncate. That bites these controls harder than most, because each one can hold a value this build does not recognise.

Each handles it, and I verified the store is genuinely untouched rather than trusting the display behaviour:

- **Entity picker** — an id absent from this Home Assistant is returned and shown as configured ("It stays configured, and the card renders without it"), not blanked. An id outside the picker's own domain filters stays selected and named.
- **Number array** — entries the schema rejects render greyed and marked ignored; removing a *different* entry still emits the unusable ones, because removal is positional.
- **Ordered multi-select** — an unknown mode renders in place as unavailable and moves and removes like any other entry.

Probed directly: reading a config containing `[10, 150, 'ten', 50]` under 1–100 bounds returns `[10, 50]` for display while the stored array remains byte-identical, and the same holds for an `armModes` list carrying `armed_custom_bypass`. A sanity assertion in the same probe confirms the readers return real data for recognised input, so "unchanged" is not passing vacuously.

## Details worth noting

The entity-id schema uses Home Assistant Core's own `valid_entity_id` rule. CodeRabbit caught the first, looser pattern accepting `sensor._motion` — worth having, since a config gate that accepts ids HA itself would reject produces a document that imports here and fails there.

The ordered multi-select's schema is `z.array(z.enum(values))` refined to reject repeats — the canonical, closed value list REVIEW.md requires, with the caller narrowing `values` to what the entity actually supports.

Number arrays are append-only and never sorted, because stored order is what `brightnessPresets` renders.

## Testing

- `npm test` — 1543 passed, 2 skipped
- `npm run lint`, `npm run typecheck` — clean
- `npm run test:coverage` — 100% patch coverage on the new files, lines and branches
- `npm run build-storybook`, `npm run build:ha:prod` — both succeed
- axe over all 24 card-configuration stories in both appearances: zero serious or critical, nothing added to **#191** / **#197** / **#204** / **#210**. One real find fixed on the way — Radix soft-blue chips measured 4.25:1 in light, so the chips take `highContrast`.

`codex` was unavailable (usage limit until 1 August); CodeRabbit ran and its five findings are resolved.

## Change status

`docs/changes/0014-universal-card-options.md` flips to **complete**, with matching flips in `docs/index.yml` and `docs/index.md`. The options spec records the universal contract, action system, detail dialog and shared controls as implemented, while the per-card option docs correctly remain "specified, not yet implemented".

## Summary by CodeRabbit

* **New Features**
  * Added shared configuration controls for selecting entities, editing numeric lists, and managing ordered multi-selections.
  * Entity selection supports search, filtering, clearing, and clear warnings for unavailable entities.
  * Numeric list editing supports validation, duplicate prevention, ordering, and removal.
  * Ordered selections support adding, reordering, removing, and preserving unavailable values.
* **Documentation**
  * Marked the universal card options contract as complete and clarified remaining card-specific documentation status.
* **Tests**
  * Added comprehensive coverage for the new controls, validation, filtering, and data-handling scenarios.

### CODE REVIEW

File: src/components/EntityPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+      </Text>
+
+      <Popover.Root open={open} onOpenChange={setOpen}>
+        <Popover.Trigger>
+          <Button size="3" variant="soft" color="gray" aria-label={label}>
+            <Text truncate>
+              {selectedEntity ? friendlyNameOf(selectedEntity) : selectedId || placeholder}
+            </Text>
+          </Button>
+        </Popover.Trigger>
```
[the remark this answers is not in the document]
Comment (fx) [reply src/components/EntityPicker.tsx:214]: Fixed — `search` is cleared whenever the popover closes, not only on selection.

Good catch on the consequence rather than the mechanism: the damage is not the stale string itself but that reopening silently re-applies a filter the user does not remember setting, so an entity that is present looks missing. Test added covering dismiss-without-selection followed by reopen.

File: src/store/configControls.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+ */
+
+/**
+ * `domain.object_id`, the only shape a Home Assistant entity id takes — the
+ * same rule Core's own `valid_entity_id` applies: lowercase alphanumerics and
+ * underscores, with no leading, trailing, or doubled underscore in either
+ * segment. Copied rather than loosened so the schema rejects exactly what Home
+ * Assistant would refuse to create.
+ */
+const ENTITY_ID_PATTERN = /^(?!.+__)(?!_)[\da-z_]+(?<!_)\.(?!_)[\da-z_]+(?<!_)$/
```
[the remark this answers is not in the document]
Comment (fx) [reply src/store/configControls.ts]: I checked this against a live Home Assistant rather than deciding between our readings, and the premise is wrong: HA's `valid_entity_id` IS the strict pattern.

From the running instance's own `homeassistant/core.py`:

```
VALID_ENTITY_ID = ^(?!.+__)(?!_)[\da-z_]+(?<!_)\.(?!_)[\da-z_]+(?<!_)$
```

So `sensor._motion` and `sensor.front__door` are rejected by Home Assistant itself, and the gate matching that is correct — a config gate looser than HA's produces documents that import here and fail there. The pattern keeps a comment recording where it comes from and that it was verified against a live instance, so this does not get re-litigated.

**Your secondary point I did act on, and it mattered more than the one you led with.** `(?<!_)` is a parse-time syntax error on engines without lookbehind — Safari before 16.4 — so the regex literal would take down the whole module rather than degrade. For a wall-tablet dashboard where the HA companion app is a realistic client, that is a hard bundle failure worth removing.

The pattern is now built from `[\da-z]+(?:_[\da-z]+)*` per segment, which forbids leading, trailing and doubled underscores structurally, with no assertions at all. Equivalence is proved rather than asserted: a test runs both the HA pattern and the replacement over a corpus of 33 cases — leading/trailing/doubled underscores, digits, single-character and empty segments, missing and extra dots, uppercase, unicode, whitespace — and asserts they agree on every one, with a guard that the corpus contains both accepted and rejected ids so it cannot pass vacuously.

File: src/store/__tests__/configControls.test.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    ['sensor.'],
+    ['sensor.a.b'],
+    // The same forms Home Assistant's own `valid_entity_id` refuses: a leading,
+    // trailing, or doubled underscore in either segment.
+    ['sensor._motion'],
+    ['sensor.motion_'],
+    ['_sensor.motion'],
+    ['sensor__aux.motion'],
+    ['sensor.front__door'],
+  ])('rejects the malformed id %s', (value) => {
```
[the remark this answers is not in the document]
Comment (fx) [reply src/store/__tests__/configControls.test.ts:44]: The tests stay as they are — they encode Home Assistant's real contract. Verified from the running instance's `homeassistant/core.py`: `VALID_ENTITY_ID` is `^(?!.+__)(?!_)[\da-z_]+(?<!_)\.(?!_)[\da-z_]+(?<!_)$`, so HA does reject `sensor._motion` and `sensor.front__door`.

The suite gained an equivalence test alongside them, comparing the implementation against that exact HA pattern across a 33-case corpus, so the contract is now pinned to its source rather than to a hand-written list of examples.

File: src/components/EntityPicker.tsx [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+   * An empty list has three causes and only one of them is the user's search.
+   * Saying "no match" while the panel is still loading, or while it is not
+   * talking to Home Assistant at all, describes the config as the problem when
+   * the connection is.
+   */
+  const emptyMessage = isLoading
+    ? 'Still loading entities from Home Assistant…'
+    : !isConnected
+      ? 'Not connected to Home Assistant — no entities to choose from.'
+      : 'No entity matches that search.'
```
[the remark this answers is not in the document]
Comment (fx) [reply src/components/EntityPicker.tsx]: Fixed — the empty state now has five branches, each true of the situation it names:

- still loading from Home Assistant
- not connected
- this Home Assistant has no entities at all
- this Home Assistant has no entities of the kind the option asks for (naming the filters)
- no entity matches that search

The fourth is the one worth having separately from the third, and it is why I did not just soften the wording: "no entities of that kind exist here" and "your filters exclude everything" point at different next actions, and the component can tell them apart, so it should. The adjacent comment now describes the branch it actually sits on.

Each branch has a test. Good catch — the misleading message was reachable from an ordinary story with no seeded entities, so it was not an edge case so much as the default.

File: src/components/EntityPicker.tsx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+      .filter((entity) => matchesFilters(entity, domains, deviceClasses))
+      .sort((a, b) => friendlyNameOf(a).localeCompare(friendlyNameOf(b)))
+  }, [entities, domains, deviceClasses])
+
+  const matches = React.useMemo(
+    () => available.filter((entity) => matchesSearch(entity, search)),
+    [available, search]
+  )
+
+  const shown = matches.slice(0, MAX_RESULTS)
```
[the remark this answers is not in the document]
Comment (fx) [reply src/components/EntityPicker.tsx:169]: Fixed — the list is not built at all while the popover is closed, and the memo returns a shared empty array so a closed picker keeps a stable reference.

On narrowing the subscription, which you implied: it cannot be narrowed. The picker's filters are predicates over domain and `device_class`, not a list of ids, so there is nothing for `useEntities()` to subscribe to more specifically — it takes the whole store map and the component re-renders on every batch Home Assistant sends. Gating the work is the available fix, not a workaround for a missing one.

Nothing is deferred past the point it is needed: `open` flips during render, so the frame that opens the popover builds the list from the entities as they are right then. There is a test for exactly the failure that would otherwise hide here — an entity renamed while the popover was closed must appear under its new name on open, not a stale one.

Pinned by a test asserting the list is never built however many entity updates arrive while closed. Verified it is not vacuous: removing the gate fails it.

Worth agreeing on the stakes rather than treating it as a micro-optimisation — this control appears in every card's config modal, and a real Home Assistant carries thousands of continuously-changing entities, so the symptom was a config form stuttering while the user typed in an unrelated field.
